src/lib/externaldata.py

```diff
@@ -74,9 +74,9 @@ def __str__(self):
                '  Checker: {checker_data}'.format(state=self.state.name,
                                                   filename=self.filename,
                                                   type=self.type.name,
-                                                  url=self.url,
-                                                  checksum=self.checksum,
-                                                  size=self.size,
+                                                  url=self.new_version.url if self.new_version else self.current_version.url,
```

Comment:
Personally I would prefer (eg):

```python3
version = self.new_version or self.current_version
...
    url=version.url,
    checksum=version.checksum,
...
```

but, not a big deal.

Reply:
Done.